### PR TITLE
Feature powerupscore

### PR DESCRIPTION
**Pull request by LeonWubben**
Adds score when powerups are activated
Points are rewarded when the powerup is activated
And points are rewarded for every gem the powerup removes.

### CODE REVIEW

**Comment by ghyl** [comment]
Firstly, Score shouldn't have its own package, it should be in the model package

File: src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
     public static final int GRID_DIMENSIONS = 8;
     private Piece[][] grid = new Piece[GRID_DIMENSIONS][GRID_DIMENSIONS];
-    private int totalScore;
+    private int tempScore;
```
Comment (ghyl) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java]: The responsibility for keeping score shouldn't be for the board. So for a future iteration we should refactor this.

File: src/main/java/nl/tudelft/ti2206/semgr8/model/piece/BombGem.java [hunk 2]
```diff
@@ -34,10 +35,15 @@ public String toString() {
...
-            board.removePiece(getPosition().getX() + i, getPosition().getY() - 1);
-            board.removePiece(getPosition().getX() + i, getPosition().getY());
-            board.removePiece(getPosition().getX() + i, getPosition().getY() + 1);
+        board.addScore(Score.SCORE_FOR_BOMB_REMOVED);
+        for (int i = -1; i <= 1; i++) {
+            for (int j = -1; j <= 1; j++) {
+                boolean isRemoved = board.removePiece(getPosition().getX() + i, getPosition().getY() + j);
+                if(isRemoved)
+                {
+                    board.addScore(Score.SCORE_PER_PIECE_REMOVED_BY_BOMB);
```
Comment (ghyl) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/piece/BombGem.java]: we don't use this bracket style, instead;
> if(isRemoved) {
>      ...
>}`

File: src/main/java/nl/tudelft/ti2206/semgr8/score/Score.java [hunk 3]
```diff
@@ -12,6 +12,20 @@
...
+
+    public static final int SCORE_PER_PIECE_REMOVED_BY_LASER = 20;
+    public static final int SCORE_FOR_LASER_REMOVED = 50;
+
+    public static final int SCORE_PER_PIECE_REMOVED_BY_BOMB = 50;
+    public static final int SCORE_FOR_BOMB_REMOVED = 50;
+
+    public static final int SCORE_PER_PIECE_REMOVED_BY_LIGHTNING = 50;
+    public static final int SCORE_PER_LIGHTNING_REMOVED = 100;
+
```
Comment (ghyl) [inline src/main/java/nl/tudelft/ti2206/semgr8/score/Score.java]: it might be better to make a seperate class for these static ints..

**Comment by ghyl** [comment]
when comments are processed, then lgtm

File: src/main/java/nl/tudelft/ti2206/semgr8/score/Score.java (hunk 3 above)
Comment (LeonWubben) [inline src/main/java/nl/tudelft/ti2206/semgr8/score/Score.java]: For now I'll leave it like this, in a future iteration we might make a class that holds all adjustable game constants.

**Comment by albert-smit** [comment]
looks good

**Comment by wroel** [comment]
lgtm

**Pull request merged**